### ISSUE TITLE
Make widget themable

### ISSUE THREAD

**Issue by LuccoJ** [martykan/forecastie#133]
With the new version of Forecastie coming with a new brighter default theme, but me preferring to switch back to the classic one, I realized that he widget ignores the setting and displays itself in the bright theme.

Additionally, whenever the app itself displays a new activity, the title bar briefly flashes in the default color before changing to the set theme.

**Comment by kn00tcn** [comment]
haha, the widget preview is the old theme (for me at least), then it turns bright blue when added

i gotta say, the fresh blue is awful, there are too many bright blue apps already (CM/aosp? phone, CM file manager, fdroid), it's just a harsh blue while the classic one was relaxing, a little hip, & most importantly solid branding that only forecastie used

how do i reproduce the title bar flash? i tried pushing refresh & dont see any fresh blue appear

**Comment by LuccoJ** [comment]
To cause the bright blue flashing, I just start the app, either by using its icon or by clicking on its widget, and the title bar is bright blue for a second, then switches to my set theme.

The widget preview is indeed the old color until you actually put it on your home screen.

**Comment by kn00tcn** [comment]
oh hmm... now i see the blue flash, which doesnt appear when you enter from recents (or rather, the screen sits on the frozen recents thumbnail until the app is ready)

**Comment by koug44** [comment]
I also would like the widget to match the app theme (Dark classic in my case)

**Comment by collectorgeneral** [comment]
Agreed, widget theme should change according to main theme.

**Comment by kn00tcn** [comment]
such a small app, maybe i should have looked at the code for fun, am i the only one that has programming (but not android) experience?

**Comment by collectorgeneral** [comment]
@kn00tcn 
I am not even a programmer 😆 
I barely remember some basic stuff about programming which I learned back at university. I am an Engineer, so I really did not learn java. If you can understand android code, then by all means, look through it!

**Comment by kn00tcn** [comment]
i doubt it's going to be android specific, most languages look quite similar, so it's just going to be an attempt to search for color usage, maybe checking what was added when the colors were first added & attempting to copy the same code over to the widget (if lucky)

**Comment by robinpaulson** [comment]
If one of you could look up the relevant code, that would be a big help (as would making a pull request with a fix...).

**Comment by ildar** [comment]
If you allow me to donate my 5 cents 😀
I suggest that app theme and widget theme should be independent. For some
reasons what you see in apps and Launcher staff quite often are different.
That's what I also see in some apps - different settings for app and widget.

**Comment by robinpaulson** [comment]
@ildar 
This is a valid point; it would require a separate item in the settings. Perhaps:
Widget theme:
===========
* Same as app
* Dark
* Light
* Night
etc.

Now that #164 has been implemented, does that make all this simpler?

Is anyone keen to start on this; a sizeable chunk of it will be a copy and paste from the app theme code.

**Comment by kn00tcn** [comment]
holy crap i never noticed how many contributors there were, i always thought it was a one guy app

i have a very unreliably vague interest in taking a look that i may not even get around to, but maybe that depends on how complicated the app theme code is, will have to take a look first

**Comment by robinpaulson** [comment]
#186 Includes relevant requests and information which we should make sure we cover here, particularly the desire for a transparent widget.

**Comment by arucard21** [comment]
I created a custom APK with the widget background set to transparent. This is a bit of a workaround until this theming functionality can be implemented, which is obviously the better solution. I just set the background of all 3 widgets to transparent. The text is still white so obviously this only works with dark wallpapers. It may not be useful to others but since I already created it for myself, it was easy to share.

The APK file can be found here: https://github.com/arucard21/forecastie/blob/master/app/release/app-release.apk
It's self-signed and I've kept the rather generic name for the file to discourage using this workaround too much.
